floris/tools/optimization/layout_optimization/layout_optimization_base.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 NREL
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# See https://floris.readthedocs.io for documentation
+
+import numpy as np
+import matplotlib.pyplot as plt
+from shapely.geometry import Polygon, LineString
+
+from ....logging_manager import LoggerBase
+
+class LayoutOptimization(LoggerBase):
+    def __init__(self, fi, boundaries, min_dist=None, freq=None):
+        self.fi = fi
+        self.boundaries = boundaries
+
+        self.boundary_polygon = Polygon(self.boundaries)
```

Comment:
Since these are derivatives of the public `boundaries` variable, should `self.boundary_polygon` be private, something like `self._boundary_polygon`?

Reply:
Updated!